### DISCUSSION TITLE
How to close all GLVis windows?

### DISCUSSION THREAD

**Discussion by benzwick** [GLVis/glvis#128] [Q&A] [answered]
Is there a recommended way to close all windows associated with a GLVis server process? I am routinely opening many plots to visualize tensor components but every time I re-run my simulation I have to manually close all these windows. Currently I just `killall glvis`. Is there a better way to do this? Even better, how difficult would it be to add a keyboard shortcut (Q say) that closes all windows?

**Comment by QiangWei-Wade** [comment]
I am wondering the same.

**Comment by tzanio** [comment]
Converting from issue to discussion...

**Comment by tzanio** [comment]
I understand the problem but unfortunately, I don't think this is possible currently. 

One option could be that you reuse the windows instead of opening new ones?

@v-dobrev and @kanye-quest -- any ideas?

**Reply by v-dobrev** [reply]
One problem is that all GLVis windows are separate processes, so sending them a message to close may be tricky.

It may be possible to query the window manager for all GLVis windows and then tell the window manager to close them.

The brute force method will be to just call `killall glvis` or equivalent, possibly skipping the parent process which is the GLVis server itself.

**Reply by publixsubfan** [reply]
I think we create new GLVis windows with `fork()` right? Maybe we could save the returned child PIDs in the master process and send them a SIGTERM when exiting the listener process.

**Comment by tzanio** [comment]
This will be addressed by https://github.com/GLVis/glvis/pull/149

**Comment by tzanio** [comment, marked as answer]
This should be resolved in master as of b513773 (Jul 9th, 2021)
